frontend/src/components/utils/MultiUserSelector.tsx

```diff
@@ -0,0 +1,249 @@
+        .sort(compareUsers);
+
+    const {
+        data: promotions,
+        status: promotionsStatus,
+        error: promotionsError,
+    } = useBetterQuery<{
+        promotions: string[];
+    }>(["promotions.list"], api.users.listPromotions, {
+        refetchOnWindowFocus: false,
+    });
+
+    useEffect(() => {
+        setPromotion({
+            value: loggedUser?.promotion || "",
+            label: loggedUser?.promotion || "",
+        });
+    }, [loggedUser]);
+
+    useEffect(() => {
+        if (!promotion) {
+            return;
+        }
+
+        setUserStatus("loading");
+        api.users
+            .list({
+                promotion: promotion.value,
+                search: searchValue,
+                page_size: 1000,
+            })
+            .then((response) => {
+                setUserStatus("success");
+                let newUsers = { ...users };
+                for (let userId in users) {
+                    if (!users[userId].selected) {
+                        delete newUsers[userId];
+                    }
+                }
+
+                response.results.forEach((user) => {
+                    if (!newUsers.hasOwnProperty(user.id))
+                        newUsers[user.id] = {
+                            selected: false,
+                            user: user,
+                        };
+                });
+
+                setUsers(newUsers);
+            })
+            .catch((error) => {
+                setUserStatus("error");
+                setUserError(error);
+            });
+
+        // Avoid to trigger infinite update by putting users as a dep
+        // eslint-disable-next-line
+    }, [promotion, searchValue]);
+
+    const addUser = (user) => {
+        setUsers((users) => ({
+            ...users,
+            ...{
+                [user.id]: {
+                    user: user,
+                    selected: true,
+                },
+            },
+        }));
+    };
+
+    const removeUser = (user) => {
+        if (user.promotion === promotion.value) {
+            setUsers((users) => ({
+                ...users,
+                ...{
+                    [user.id]: {
+                        user: user,
+                        selected: false,
+                    },
+                },
+            }));
+        } else {
+            setUsers((users) => {
+                let newUsers = { ...users };
+                delete newUsers[user.id];
+                return newUsers;
+            });
+        }
+    };
+
+    useEffect(() => {
+        onChange(selectedUsers);
+    }, [selectedUsers, onChange]);
+
+    return userStatus === "error" ? (
+        <ErrorMessage>{userError}</ErrorMessage>
+    ) : promotionsStatus === "error" ? (
+        <ErrorMessage>{promotionsError}</ErrorMessage>
+    ) : promotionsStatus === "loading" ? (
+        <Loading />
+    ) : (
+        <Row className={"border bg-white"} style={{ height: "80vh" }}>
```

Comment:
If you need a `Row`, put it in a `Container`.
https://stackoverflow.com/questions/19063976/bootstrap-grid-do-i-need-a-container

It works fine for this example, but having the container will help with the reusability.